optuna/_gp/acqf.py

```diff
@@ -54,12 +55,8 @@ def logehvi(
     # Check the implementations here:
     # https://github.com/pytorch/botorch/blob/v0.13.0/botorch/utils/safe_math.py
     # https://github.com/pytorch/botorch/blob/v0.13.0/botorch/acquisition/multi_objective/logei.py#L146-L266
-    _EPS = torch.tensor(1e-12, dtype=torch.float64)  # NOTE(nabenabe): grad becomes nan when EPS=0.
-    diff = torch.maximum(
-        _EPS,
-        torch.minimum(Y_post[..., None, :], non_dominated_box_upper_bounds)
-        - non_dominated_box_lower_bounds,
-    )
+    diff = Y_post.unsqueeze(-2) - non_dominated_box_lower_bounds
+    diff.clamp_(min=torch.tensor(_EPS).double(), max=non_dominated_box_intervals)
```

Comment:
There is a small error.

```suggestion
    diff.clamp_(min=torch.tensor(_EPS, dtype=torch.float64), max=non_dominated_box_intervals)
```

Reply:
Isn't it indeed the same thing?
I am very flexible regarding here, but I asked the question out of my curiosity!

Reply:
They are different. It's a very trivial difference, but since we can maintain the original behavior, I prefer `dtype=torch.float64`.
```
>>> torch.tensor(1e-12).double().numpy()
array(9.99999996e-13)
>>> torch.tensor(1e-12, dtype=torch.float64).numpy()
array(1.e-12)
```

Reply:
You are right.
I observed that `torch.tensor` first casts `1e-12` to `float32`, which halves the original precision.
Thank you for pointing it out!